client/src/components/DocumentForm.tsx

```diff
@@ -0,0 +1,366 @@
+import dayjs from "dayjs";
+import { FC, useEffect, useState } from "react";
+import "../styles/DocumentForm.scss";
+import {
+  Coordinates,
+  Document,
+  documentFormDefaults,
+  DocumentForm as DocumentFormType,
+  DocumentType,
+  Link,
+  Scale,
+  ScaleType,
+  Stakeholder,
+} from "../utils/interfaces";
+import LinksTable from "./LinksTable";
+import ProgressBar from "./ProgressBar";
+import SearchBar from "./SearchBar";
+
+interface DocumentFormProps {
+  newPos: Coordinates;
+  onClose: () => void;
+  onSubmit: (newDocument: DocumentFormType) => void;
+  documents: Document[];
+  closePositionView: () => void;
+  editDocument: Document | null;
+}
+
+const stakeholdersOptions = [
+  { value: Stakeholder.Lkab, label: "LKAB" },
+  { value: Stakeholder.KirunaKommun, label: "Kiruna kommun" },
+  { value: Stakeholder.Residents, label: "Residents" },
+  { value: Stakeholder.WhiteArkitekter, label: "White Arkitekter" },
+];
+
+const DocumentForm: FC<DocumentFormProps> = ({
+  newPos,
+  onClose,
+  onSubmit,
+  documents,
+  closePositionView,
+  editDocument,
+}) => {
+  const [page, setPage] = useState<number>(1);
+  const [document, setDocument] = useState<DocumentFormType>(
+    editDocument || documentFormDefaults
+  );
+  const [tableLinks, setTableLinks] = useState<Link[]>(document?.links || []);
+
+  useEffect(() => {
+    setDocument((prev) => ({
+      ...prev,
+      coordinates: { latitude: newPos.latitude, longitude: newPos.longitude },
+    }));
+  }, [newPos]);
+
+  const onCheckboxChange = (event: {
+    target: { value: string; checked: boolean };
+  }) => {
+    const { value, checked } = event.target;
+
+    setDocument(
+      (previousDoc) =>
+        ({
+          ...previousDoc,
+          stakeholders: checked
+            ? [...(previousDoc.stakeholders ?? []), value]
+            : previousDoc.stakeholders?.filter((sh) => sh != value) || [],
+        }) as Document
+    );
+  };
+
+  const onDateChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    setDocument((prev) => ({
+      ...prev,
+      issuanceDate: value ? dayjs(value) : (prev.issuanceDate ?? undefined),
+    }));
+  };
+
+  const handleFormSubmit = (ev: React.FormEvent) => {
+    ev.preventDefault();
+    onSubmit({
+      ...document,
+      links: tableLinks,
+    });
+
+    resetForm();
+  };
+
+  const resetForm = () => {
+    setDocument(documentFormDefaults);
+    setPage(1);
+    closePositionView();
+  };
+
+  return (
+    <form
+      className={`document modal page-${page}`}
+      onSubmit={(ev) => {
+        ev.preventDefault();
+        if (page === 1) {
+          // Passa alla pagina successiva solo se la validazione è passata
```

Comment:
Is this part still to be implemented?

Reply:
I left this unchanged from the previous impl @LorenzoRicci26

Reply:
Ok